lib/rules/no-unused-message-ids.ts

```diff
@@ -99,21 +101,27 @@ const rule = {
 
           const reportMessagesAndDataArray =
             collectReportViolationAndSuggestionData(reportInfo);
-          for (const { messageId } of reportMessagesAndDataArray.filter(
-            (obj) => obj.messageId,
-          )) {
+          for (const messageId of reportMessagesAndDataArray
+            .map((obj) => obj.messageId)
+            .filter((messageId) => !!messageId)) {
             const values =
               messageId.type === 'Literal'
                 ? [messageId]
-                : findPossibleVariableValues(messageId, scopeManager);
+                : findPossibleVariableValues(
+                    messageId as Identifier,
+                    scopeManager,
+                  );
```

Comment:
Would prefer type checks over casting, and just bail out if it's not the right type.

```suggestion
                : typeof messageId === 'Identifier' ? findPossibleVariableValues(
                    messageId,
                    scopeManager,
                  ): [];
```

Reply:
Updated